scr/src/main/java/org/apache/felix/scr/impl/BundleComponentActivator.java

```diff
@@ -98,133 +98,82 @@
 
     private static class ListenerInfo implements ServiceListener
     {
-        private Map<Filter, List<ExtendedServiceListener<ExtendedServiceEvent>>> filterMap = new HashMap<>();
+        List<ExtendedServiceListener<ExtendedServiceEvent>> listeners = new ArrayList<>();
```

Comment:
Shouldn't this just be a CopyOnWriteArrayList?

Reply:
It has been a while I made these changes, so I am not too sure, but I believe I mimicked the current approach, but with a List rather than a Map. Looking at the places where the list is modified, it looks similar to what a CopyOnWriteArrayList would do.

I don't mind changing it, though I am not familiar with possible additional reasons why it was done in this way.